File: payment-service/src/main/java/com/hedvig/paymentservice/services/trustly/dto/DirectDebitOrderInfo.java
package com.hedvig.paymentservice.services.trustly.dto;

import com.hedvig.paymentservice.web.dtos.RegisterDirectDebitRequestDTO;
import lombok.Value;

@Value
public class DirectDebitOrderInfo {

  String memberId;
  String personalNumber;
  String firstName;
  String lastName;
  String triggerId;
  boolean redirectingToBotService;

  public DirectDebitOrderInfo(DirectDebitRequest request, boolean isRedirectingToBotService) {
    this.memberId = request.getMemberId();
    this.personalNumber = request.getSsn();
    this.firstName = request.getFirstName();
    this.lastName = request.getLastName();
    this.triggerId = request.getTriggerId();
    this.redirectingToBotService = isRedirectingToBotService;
  }

  public DirectDebitOrderInfo(
      String memberId,
      RegisterDirectDebitRequestDTO request,
      boolean isRedirectingToBotService) {
    this.memberId = memberId;
    this.personalNumber = request.getPersonalNumber();
    this.firstName = request.getFirstName();
    this.lastName = request.getLastName();
    this.triggerId = null;
    this.redirectingToBotService = isRedirectingToBotService;
  }
}

File: payment-service/src/main/java/com/hedvig/paymentservice/services/trustly/dto/DirectDebitOrderInfo.kt
package com.hedvig.paymentservice.services.trustly.dto

import com.hedvig.paymentservice.serviceIntergration.memberService.dto.Member
import com.hedvig.paymentservice.web.dtos.RegisterDirectDebitRequestDTO

data class DirectDebitOrderInfo(

    val memberId: String,
    val personalNumber: String,
    val firstName: String,
    val lastName: String,
    val triggerId: String? = null,
    val redirectingToBotService: Boolean = false
) {
  constructor(request: DirectDebitRequest, isRedirectingToBotService: Boolean) :
    this(
      request.memberId,
      request.ssn,
      request.firstName,
      request.lastName,
      request.triggerId,
      isRedirectingToBotService
    )

  constructor(
      memberId: String,
      request: RegisterDirectDebitRequestDTO,
      isRedirectingToBotService: Boolean
  ) : this(memberId, request.personalNumber, request.firstName, request.lastName, null, isRedirectingToBotService)

  companion object {
    fun fromMember(m: Member): DirectDebitOrderInfo {
      return DirectDebitOrderInfo(
        m.memberId,
        m.ssn,
        m.firstName,
        m.lastName,
        null,
        false
      )
    }
  }
}
